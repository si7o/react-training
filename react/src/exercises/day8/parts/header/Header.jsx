import React from "react";
import { AppBar, Box } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import HeaderLogo from "./HeaderLogo";
import UserInfo from "./UserInfo";
import SearchBox from "./SearchBox";
import { useLocation } from "react-router";

// styles using makeStyles & theme
const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    alignItems: "center",
    flexDirection: "row",
    paddingRight: 20,
  },
}));

const Header = () => {
  const classes = useStyles();
  const location = useLocation();

  const title = location.state?.title || "Dashboard";

  return (
    <AppBar color="transparent" className={classes.root}>
      <HeaderLogo text="Leads Builder" />
      <Box
        display="flex"
        alignItems="flex-start"
        ml={1}
        fontSize={18}
        color="#a8a8a8"
        fontWeight="500"
        fontFamily="Verdana"
      >
        {title}
      </Box>
      <SearchBox />
      <UserInfo username="User Name" />
    </AppBar>
  );
};

export default Header;
