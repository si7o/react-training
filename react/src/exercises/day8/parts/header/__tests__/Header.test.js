import React from "react";
import { render } from "@testing-library/react";

import { MemoryRouter } from "react-router";

import Header from "../Header";

describe("<Header>", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
