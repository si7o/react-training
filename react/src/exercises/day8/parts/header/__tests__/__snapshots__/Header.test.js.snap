// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorTransparent makeStyles-root-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="div-root-2"
    >
      Leads Builder
    </div>
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      Dashboard
    </div>
    <div
      class="MuiInputBase-root makeStyles-inputRoot-4 MuiInputBase-adornedEnd"
    >
      <input
        aria-label="search"
        class="MuiInputBase-input makeStyles-inputInput-5 MuiInputBase-inputAdornedEnd"
        placeholder="Search"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        color="#3381f7"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      class="makeStyles-root-6"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-7 MuiAvatar-colorDefault"
      >
        U
      </div>
      <span
        class="makeStyles-username-8"
      >
        User Name
      </span>
    </div>
  </header>
</DocumentFragment>
`;
